geopandas/tests/test_geodataframe_to_file.py

```diff
@@ -0,0 +1,217 @@
+        (-73.5540185061397, 45.5084409343852)
+    )),
+    LineString((
+        (-73.5539986525799, 45.5084323044531),
+        (-73.5535801792994, 45.5089539203786),
+        (-73.5541107525234, 45.5091983609661)
+    ))
+]
+
+city_hall_entrance = Point(-73.553785, 45.508722)
+city_hall_balcony = Point(-73.554138, 45.509080)
+city_hall_council_chamber = Point(-73.554246, 45.508931)
+
+point_3D = Point(-73.553785, 45.508722, 300)
+
+
+@pytest.fixture(params=[
+    # Points
+    GeoDataFrame(
+        {'a': [1, 2]},
+        crs={'init': 'epsg:4326'},
+        geometry=[city_hall_entrance, city_hall_balcony]
+    ),
+    # MultiPoints
+    GeoDataFrame(
+        {'a': [1, 2]},
+        crs={'init': 'epsg:4326'},
+        geometry=[
+            MultiPoint([
+                city_hall_balcony,
+                city_hall_council_chamber]),
+            MultiPoint([
+                city_hall_entrance,
+                city_hall_balcony,
+                city_hall_council_chamber]
+            )]
+    ),
+    # LineStrings
+    GeoDataFrame(
+        {'a': [1, 2]},
+        crs={'init': 'epsg:4326'},
+        geometry=city_hall_walls
+    ),
+    # LineStrings and MultiLineStrings
+    GeoDataFrame(
+        {'a': [1, 2]},
+        crs={'init': 'epsg:4326'},
+        geometry=[MultiLineString(city_hall_walls), city_hall_walls[0]]
+    ),
+    # MultiLineStrings
+    GeoDataFrame(
+        {'a': [1, 2]},
+        crs={'init': 'epsg:4326'},
+        geometry=[
+            MultiLineString(city_hall_walls),
+            MultiLineString(city_hall_walls)
+        ]
+    ),
+    # Polygons
+    GeoDataFrame(
+        {'a': [1, 2]},
+        crs={'init': 'epsg:4326'},
+        geometry=[city_hall_boundaries, vauquelin_place]
+    ),
+    # MultiPolygon and Polygon
+    GeoDataFrame(
+        {'a': [1, 2]},
+        crs={'init': 'epsg:4326'},
+        geometry=[
+            MultiPolygon((city_hall_boundaries, vauquelin_place)),
+            city_hall_boundaries
+        ]
+    ),
+    # MultiPolygon
+    GeoDataFrame(
+        {'a': [1]},
+        crs={'init': 'epsg:4326'},
+        geometry=[MultiPolygon((city_hall_boundaries, vauquelin_place))]
+    ),
+    # Null geometry and Point
+    GeoDataFrame(
+        {'a': [1, 2]},
+        crs={'init': 'epsg:4326'},
+        geometry=[None, city_hall_entrance]
+    ),
+    # Null geometry and 3D Point
+    GeoDataFrame(
+        {'a': [1, 2]},
+        crs={'init': 'epsg:4326'},
+        geometry=[None, point_3D]
+    ),
+    # Null geometries only
+    GeoDataFrame(
+        {'a': [1, 2]},
+        crs={'init': 'epsg:4326'},
+        geometry=[None, None]
+    )
+
+])
+def geodataframe(request):
+    return request.param
+
+@pytest.fixture(params=[
+    # Points and MultiPoints
+    # It may be a bug in Fiona or in 'ESRI Shapefile' driver:
+    # a gdf with Polygons and MultiPolygons can be written to shapefiles
+    # while a gdf with Points and MultiPoints cannot...
```

Comment:
I think this is expected behaviour. https://www.gdal.org/drv_shapefile.html mentions LineString/MultiLinestring and Polygon/MultiPolygon, but not points.